src/components/auth/HeaderProfileDropdown.tsx

```diff
@@ -0,0 +1,67 @@
+import { Avatar, AvatarFallback, AvatarImage } from "../ui/avatar"
+import { DropdownMenu, DropdownMenuTrigger, DropdownMenuContent, DropdownMenuSeparator } from "../ui/dropdown-menu"
+import { Button } from "../ui/button";
+import { ChevronRightIcon } from "@heroicons/react/24/solid";
+import api from "../../api/backend";
+import { useContext, useState } from "react";
+import { ClipLoader } from "react-spinners";
+import SessionContext from "../../contexts/SessionContext";
+
+export const HeaderProfileDropdown = () => {
+  const [loggingOut, setLoggingOut] = useState(false);
+
+  const { user, setSignedIn } = useContext(SessionContext);
+
+  const logout = async () => {
+    setLoggingOut(true);
+
+    fetch(`${api.OIDC_LOGOUT_URL}/`, {
+      method: "POST", credentials: "include", headers: {
+        "X-CSRFToken": api.getCSRFToken()
+      }
+    }).then(() => {
+      setSignedIn(false);
+      setLoggingOut(false);
+    }).catch((e) => {
+      console.error(e);
+    });
```

Comment:
What do you think of moving this logic to a method in authService?

Reply:
I agree!